distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Pattern.enso

```diff
@@ -352,8 +353,8 @@ type Match_Iterator
        Also returns the next iterator, if there was a match.
     next : Match_Iterator_Value
     next self =
-        regex_result = self.pattern.internal_regex_object.exec self.input self.cursor
-        case regex_result.isMatch of
+        regex_result = if self.cursor > self.input.char_vector.length then Nothing else self.pattern.internal_regex_object.exec self.input self.cursor
```

Comment:
Why it's not?
```suggestion
        regex_result = if self.cursor >= self.input.char_vector.length then Nothing else self.pattern.internal_regex_object.exec self.input self.cursor
```

I assume you already tested it well enough, there is a good reason, so I'm not suggesting to change. Just would like to understand it.

If it's `== length` it's already after last char so seems enough to end?

Reply:
`== length` gets you the empty match to meet the bits.
`"".find_all ".*"` starts at cursor 0 and returns a match of `""`